pyvista/core/common.py

```diff
@@ -718,17 +718,8 @@ def transform(self, trans):
             raise ValueError(
                 "Transform element (3,3), the inverse scale term, is zero")
 
-        # Normalize the transformation to account for the scale
-        t /= t[3, 3]
+        transformations.apply_transformation_to_points(t, self.points, inplace=True)
```

Comment:
Regarding:
```py
t /= t[3, 3]
```

It's still necessary to keep this here as this is the overall "scaling factor" associated with the transformation matrix that we don't account for in `apply_transformation_to_points`.  In fact, we throw away the bottom row with:
```py
new_points = (transformation[np.newaxis, :, :] @ points_2.T)[0, :3, :].T
```
For example:
```py
>>> transformation = np.diag(np.ones((4)))
>>> transformation[3, 3] = 10
>>> new_points = (transformation[np.newaxis, :, :] @ points_2.T)
array([[[ 0.6934589 ,  0.43539177,  0.55437903, ...,  0.77460702,
          0.27007125,  0.47950803],
        [ 0.47708465,  0.79653964,  0.24781227, ...,  0.92809209,
          0.91483141,  0.1032954 ],
        [ 0.88495622,  0.8953158 ,  0.80401478, ...,  0.47226599,
          0.22688701,  0.65865268],
        [10.        , 10.        , 10.        , ..., 10.        ,
         10.        , 10.        ]]])
```

If we simply exclude this last row we get:
```
array([[0.6934589 , 0.47708465, 0.88495622],
       [0.43539177, 0.79653964, 0.8953158 ],
       [0.55437903, 0.24781227, 0.80401478],
       ...,
       [0.77460702, 0.92809209, 0.47226599],
       [0.27007125, 0.91483141, 0.22688701],
       [0.47950803, 0.1032954 , 0.65865268]])
```

When the actual scaled points array should be:
```
array([[0.06934589, 0.04770847, 0.08849562],
       [0.04353918, 0.07965396, 0.08953158],
       [0.0554379 , 0.02478123, 0.08040148],
       ...,
       [0.0774607 , 0.09280921, 0.0472266 ],
       [0.02700713, 0.09148314, 0.0226887 ],
       [0.0479508 , 0.01032954, 0.06586527]])
```

It's possible to get the scaling factor from this row, but it's much easier just to prescale the transformation matrix.  See the following suggestion.
```suggestion
        t /= t[3, 3]
        transformations.apply_transformation_to_points(t, self.points, inplace=True)
```

Reply:
Thanks for the reference. It makes sense to predivide the transformation by the scale factor. I put this into `apply_transformation_to_points` instead of in this filter method.